cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -68,20 +69,21 @@ func CreateAdminAndKubeletKubeConfig(masterEndpoint, pkiDir, outDir string) erro
 	}
 
 	// User admin should have full access to the cluster
-	if err := createKubeConfigFileForClient(masterEndpoint, "admin", outDir, caCert, caKey); err != nil {
-		return fmt.Errorf("couldn't create a kubeconfig file for admin: %v", err)
+	if err := createKubeConfigFileForClient(masterEndpoint, AdminKubeConfigFileName, []string{"system:masters"}, outDir, caCert, caKey); err != nil {
```

Comment:
Maybe some constants for `system:masters` and `system:nodes` if there aren't already? Can you check?